README.md

```diff
@@ -2,51 +2,71 @@
-    - `task_params_json` is a JSON string containing task specific parameters
-    - will only be called once with given `task_id`
-    - can assume `task_id/constants.RESOURCES` contains all the resources provided by task creator
-  - `NextSubtask`
-    - takes one argument `task_id`
-    - can assume `CreateTask` was called earlier with the same `task_id`
-    - returns `subtask_id` which can be an arbitrary string but the same string cannot be returned more than once
-    - also returns `subtask_params_json` which is the JSON string containing subtask specific parameters
-  - `Verify`
-    - takes two arguments `task_id` and `subtask_id` which specify which subtask results should be verified
-    - will be called with only valid `task_id` and `subtask_id` values
-    - returns a boolean indicating whether results passed the verification or not
-    - for successfully verified subtasks it most likely should also perform merging the partial results into the final one
-- provider side
-  - `Compute`
-    - takes `task_id`, `subtask_id`, `subtask_params_json`
-    - can assume the following directoees exist under `work_dir`
-      - `task_id`
-      - `task_id/constants.NETWORK_RESOURCES`
-    - can assume that under `task_id/constants.NETWORK_RESOURCES` are the resources specified corresponding `NextSubtask` call
-    - generates a single `result.zip` file under `work_dir/task_id` (subject likely to change) which is the result that will be sent back to the requestor
-- both sides
-  - `Benchmark`
-    - takes no arguments
-    - reteurns a score which indicates how efficient the machine is for this type of tasks
-    - shouldn't take much time (preferably less than a minute for medium range machines)
+The API is divided into two independent parts - requestor and provider.
 
+## Requestor
+For requestor the app should implement a long running RPC service which implements the `RequestorApp` interface from the proto files. The app should assume it will have access to a single directory (let's call it `work_dir`).
+
+### RPC methods
+
+- `CreateTask`
+  - takes two arguments `task_id` and `task_params_json`
+  - can assume that the following directories exist under `work_dir`
+    - `task_id`
+    - `task_id/constants.RESOURCES`
+    - `task_id/constants.NETWORK_RESOURCES`
+    - `task_id/constants.RESULTS`
+    - `task_id/constants.NETWORK_RESULTS`
+  - should treat `work_dir/task_id` as the working directory for the given task
+  - `task_params_json` is a JSON string containing task specific parameters
+  - will only be called once with given `task_id`
+  - can assume `task_id/constants.RESOURCES` contains all the resources provided by task creator
+- `NextSubtask`
+  - takes one argument `task_id`
+  - can assume `CreateTask` was called earlier with the same `task_id`
+  - returns `subtask_id` which can be an arbitrary string but the same string cannot be returned more than once
```

Comment:
I think if we make `Compute` return the path to the result we won't need to rely on `subtask_id` begin a directory name and this can be lifted.
Updated the readme for now anyway.